src/users/views.py

```diff
@@ -170,15 +168,12 @@ def get_object(self):
     def post (self,request):
 
         profile_id = self.request.user.profile.id
-        profile = get_object_or_404(Profile, id=profile_id)
-        try:
-            new_mentor = get_object_or_404(Mentor, profile__user=self.request.user)
-            new_mentor.active = True
-        except:
-            new_mentor = Mentor(
-                profile = profile,
-                active = True,
-            )
-            new_mentor.save()
-
-        return Response(MentorSerializer(new_mentor).data)
+        profile = Profile.objects.get(id=profile_id)
+        mentor_request = Mentor.objects.filter(profile__user=self.request.user)
+        if not mentor_request.exists():
+            mentor = Mentor(profile=profile, active=True)
+        else:
+            mentor = mentor_request[0]
```

Comment:
actually I made a mistake.  `mentor = mentor_request.first()` is technically faster and more correct.